agent/consul/rpc.go

```diff
@@ -542,18 +543,38 @@ func (s *Server) blockingQuery(queryOpts structs.QueryOptionsCompat, queryMeta s
 	timeout = time.NewTimer(queryTimeout)
 	defer timeout.Stop()
 
+	// REVIEW: The spec for this placed the atomic inc for queries_blocking inside the RUN_QUERY statement
+	// however, there are a combination of factors that I believe should mean we should place it before.
+	// - `goto RUN_QUERY` on L528 above means that we'd instrument a non-blocking query as blocking for its duration.
+	// - using defer dec on L554 would mean that we inc the counter multiple times as the blocking query waits, while
+	//   we only dec on the return of blockingQuery(). We'd need swap out defer for a dec on retry on L616 as well as
+	// 	 on the return at L621
+	// -> There I think it's simpler if we don't instrument the RUN_QUERY statement and continue to use the
+    // 	  end of s.blockingQuery()'s setup phase as the point of measurement.
+
+	// atomic inc our gauge of blockingQueries
+	atomic.AddUint64(&s.queriesBlocking, 1)
+	// atomic dec when we return from blockingQuery()
+	defer atomic.AddUint64(&s.queriesBlocking, -1)
+	// set gauge directly to a float32 of our Server's queriesBlocking
+	metrics.SetGauge([]string{"rpc", "queries_blocking"}, float32(s.queriesBlocking))
```

Comment:
I think we need to use `atomic.LoadUint64` here, because even though writing is atomic, reading is not and there could be a write in between.

Reply:
Oh, really good catch - it's easy to forget sometimes. Fixed in recent push